torch/csrc/jit/python/python_sugared_value.cpp

```diff
@@ -765,6 +765,9 @@ std::shared_ptr<SugaredValue> PythonClassValue::attr(
     const std::string& field) {
   // Resolve values from the Python object first (e.g. for static methods on
   // this type, resolve them as functions)
+  if (auto* fn = type_->findStaticMethod(field)) {
+    return std::make_shared<FunctionValue>(fn);
```

Comment:
could you explain why we are doing this?

Reply:
Sure. Let's say we have a TorchBind type `TorchBindClass`. The name `torch.classes.namespace.TorchBindClass` returns an instance of `ScriptClass` with `TorchBindClass`'s JIT type as its `class_type_` member. When compiling a *TorchScript* function that contains `torch.classes.namespace.TorchBindClass`, this is going to be resolved to a regular Python class, so a `PythonClassValue` will be created for it (see line 1121). If `type_->findStaticMethod(field)` is not null (where `type_` is set to `script_class.class_type_` on line 1121), it must be a TorchBind class (for now). In that case, we want to use the static method on the JIT type instead looking one up on the Python object (which we won't be able to compile).